packages/mds-db/stops.ts

```diff
@@ -0,0 +1,38 @@
+import { Recorded, Stop, UUID } from '@mds-core/mds-types'
+import { now } from '@mds-core/mds-utils'
+
+import schema from './schema'
+
+import { vals_sql, cols_sql, vals_list, logSql } from './sql-utils'
+
+import { getWriteableClient, getReadOnlyClient } from './client'
+
+export async function writeStop(stop: Stop): Promise<Recorded<Stop>> {
+  const client = await getWriteableClient()
+  const sql = `INSERT INTO ${schema.TABLE.stops} (${cols_sql(schema.TABLE_COLUMNS.stops)}) VALUES (${vals_sql(
+    schema.TABLE_COLUMNS.stops
+  )}) RETURNING *`
+  const values = vals_list(schema.TABLE_COLUMNS.stops, { ...stop, recorded: now() })
+  await logSql(sql, values)
+  const {
+    rows: [recorded_stop]
+  }: { rows: Recorded<Stop>[] } = await client.query(sql, values)
+  return { ...stop, ...recorded_stop }
+}
+
+export async function readStop(stop_id: UUID): Promise<Recorded<Stop>> {
+  const client = await getReadOnlyClient()
+  const sql = `SELECT * FROM ${schema.TABLE.stops} WHERE ${schema.COLUMN.stop_id} = '${stop_id}'`
+  const {
+    rows: [recorded_stop]
```

Comment:
Ok what TS/JS language feature is this? I know what the square brackets do when they're describing the key, but I'm not sure what they mean when it's the value like this.

Reply:
https://www.typescriptlang.org/docs/handbook/variable-declarations.html#destructuring

Reply:
Basically, we're destructuring the `rows` array from the SQL response, and pulling the first returned member into the `recorded_stop` variable.